### PR TITLE
Swap bool_ enum for stdbool.h

### PR DESCRIPTION
**Pull request by dan4thewin**
Following the example of coreMQTT, this PR replaces the internal bool_ enum with use of stdbool.h.

### CODE REVIEW

**Review by aggarw13 — COMMENTED** [review]
A CI check for testing build with the `stbool.readme` would be good to add. We have the check present in `coreMQTT` [here](https://github.com/FreeRTOS/coreMQTT/blob/main/.github/workflows/ci.yml#L150-L164)

**Comment by dan4thewin** [comment]
> A CI check for testing build with the `stbool.readme` would be good to add. We have the check present in `coreMQTT` [here](https://github.com/FreeRTOS/coreMQTT/blob/main/.github/workflows/ci.yml#L150-L164)

Added

**Review by aggarw13 — APPROVED** [review]

File: source/include/stdbool.readme [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ *
+ *    1) Copy this file into a directory that is in your compiler's include path.
+ *       The directory must be part of the include path for system header files,
+ *       for example passed using gcc's "-I" or "-isystem" options.
+ *
+ *    2) Rename the copied file stdbool.h.
+ *
+ */
+
+#ifndef __cplusplus
```
Comment (gkwicker) [inline source/include/stdbool.readme:19]: Can we use [the standard way to test for C99 compatibility in C code](https://en.wikipedia.org/wiki/C99#Version_detection) here?

```c
#if __STDC_VERSION__ < 199901L
/* _Bool was introduced in C99. */
    #define bool     int
    #define false    0
    #define true     1
#endif

```

**Review by gkwicker — CHANGES_REQUESTED** [review]
One question - see comment

File: source/include/stdbool.readme (hunk 1 above)
Comment (dan4thewin) [reply source/include/stdbool.readme:19]: We had something like that in coreMQTT, but Richard spoke out against it (you can see the thread in Slack).  The solution, as implemented in coreMQTT, was the outcome of that discussion.  This PR seeks symmetry with the coreMQTT solution.

**Review by gkwicker — APPROVED** [review]

**Pull request merged**